### PR TITLE
tiny pro springcloud 模板

### PR DESCRIPTION
**Pull request by wjiangwang**
# PR
新增  tiny pro springcloud 模板
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-cli/blob/main/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/toolkits/pro/src/lib/init.ts [hunk 1]
```diff
@@ -151,7 +161,21 @@ const createDatabase = async (answers: ProjectInfo) => {
 
   // 读取sql文件、新建表
   const serverPath = utils.getDistPath(`${name}/server`);
-  const databaseSqlDir = path.join(serverPath, 'app', 'database');
+  let databaseSqlDir = '';
+
+  switch (true) {
+    case serverFramework === ServerFrameworks.EggJs:
+    default:
+      databaseSqlDir = path.join(serverPath, 'app/database');
```
Comment (hello-yezi) [inline packages/toolkits/pro/src/lib/init.ts:167]: default建议放在最后

File: packages/toolkits/pro/template/server/springCloud/gateway/src/main/java/com/huawei/tiny/gateway/config/AuthorizeFilter.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import java.util.Arrays;
+
+@Component
+public class AuthorizeFilter implements GlobalFilter, Ordered {
+
+  private String[] skipAuthUrl = {
+    "/v1/user/login",
+    "/v1/user/register"
+  };
```
Comment (hello-yezi) [inline packages/toolkits/pro/template/server/springCloud/gateway/src/main/java/com/huawei/tiny/gateway/config/AuthorizeFilter.java:24]: 这里是对象不是数组？

File: packages/toolkits/pro/src/lib/init.ts [hunk 3]
```diff
@@ -151,7 +161,21 @@ const createDatabase = async (answers: ProjectInfo) => {
 
   // 读取sql文件、新建表
   const serverPath = utils.getDistPath(`${name}/server`);
-  const databaseSqlDir = path.join(serverPath, 'app', 'database');
+  let databaseSqlDir = '';
+
+  switch (true) {
```
Comment (h-ivy) [inline packages/toolkits/pro/src/lib/init.ts]: switch true？

File: packages/toolkits/pro/src/lib/init.ts [hunk 4]
```diff
@@ -151,7 +161,21 @@ const createDatabase = async (answers: ProjectInfo) => {
...
+    case serverFramework === ServerFrameworks.EggJs:
+    default:
```
Comment (h-ivy) [inline packages/toolkits/pro/src/lib/init.ts]: default放最后面，咋放到中间

File: packages/toolkits/pro/template/server/springCloud/server/src/main/java/com/huawei/tiny/server/util/Result.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  public T getData() {
+    return data;
+  }
+
+  public void setData(T data) {
+    this.data = data;
+  }
+
+  public static <T> ResponseEntity<Result<T>> success(T data) {
+    return success(data,"");
```
Comment (hello-yezi) [inline packages/toolkits/pro/template/server/springCloud/server/src/main/java/com/huawei/tiny/server/util/Result.java]: 格式化

File: packages/toolkits/pro/src/lib/init.ts (hunk 1 above)
Comment (wjiangwang) [reply packages/toolkits/pro/src/lib/init.ts:167]: 已修改

File: packages/toolkits/pro/template/server/springCloud/server/src/main/java/com/huawei/tiny/server/util/Result.java (hunk 5 above)
Comment (wjiangwang) [reply packages/toolkits/pro/template/server/springCloud/server/src/main/java/com/huawei/tiny/server/util/Result.java]: 已修改

File: packages/toolkits/pro/template/server/springCloud/gateway/src/main/java/com/huawei/tiny/gateway/config/AuthorizeFilter.java (hunk 2 above)
Comment (wjiangwang) [reply packages/toolkits/pro/template/server/springCloud/gateway/src/main/java/com/huawei/tiny/gateway/config/AuthorizeFilter.java:24]: 是一个数组   java的数组初始化就是个花括号包起来
image

File: packages/toolkits/pro/src/lib/init.ts (hunk 3 above)
Comment (wjiangwang) [reply packages/toolkits/pro/src/lib/init.ts]: 已修改

File: packages/toolkits/pro/src/lib/init.ts (hunk 4 above)
Comment (wjiangwang) [reply packages/toolkits/pro/src/lib/init.ts]: 已修改

**Pull request merged**